### PR TITLE
Frontend/development

### PR DESCRIPTION
**Pull request by kutaysaran**
Frontend is ready for the milestone.

### CODE REVIEW

File: prediction-polls/frontend/src/Pages/Auth/SignIn/index.jsx [hunk 1]
```diff
@@ -97,13 +104,20 @@ function SignIn() {
...
-      }
+      const response = await fetch(process.env.REACT_APP_BACKEND_LINK+'/login', requestOptions);
+      const data = await response.json();
+    
+    if (response.status === 201 && data.accessToken && data.refreshToken) {
+      localStorage.setItem('accessToken', data.accessToken); 
+      localStorage.setItem('refreshToken', data.refreshToken);
+      navigate("/feed");
+    } else {
+      console.log('Login failed', data);
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/Pages/Auth/SignIn/index.jsx:115]: remove log for production

File: prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx [hunk 2]
```diff
@@ -1,47 +1,58 @@
...
-    e.preventDefault();
+  const handleSubmit = async (values) => {
     try {
-      let res = await fetch("http://localhost:8000/signup", {
+      const formattedValues = { ...values };
+    if (values.birthday) {
+      if (typeof values.birthday.format === 'function') {
+        formattedValues.birthday = formatDate(values.birthday.toDate());
+      } else {
+        console.error('Birthday is not a moment object');
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx:31]: remove log.

File: prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx [hunk 3]
```diff
@@ -1,47 +1,58 @@
...
+        navigate("/auth/sign-in");
       } else {
-        setMessage("Some error occured");
+        setMessage("Username should be unique!");
+        setUsernameStatus("error");
+        setUsernameHelp("Username should be unique");
       }
     } catch (err) {
-      console.log(err);
+      console.error(err);
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx:54]: remove log

File: prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx [hunk 4]
```diff
@@ -1,47 +1,58 @@
...
+      //setMessage("An error occurred. Please try again later.");
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/Pages/Auth/SignUp/index.jsx:55]: remove commented code

File: prediction-polls/frontend/src/api/requests/logout.jsx [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+      body: JSON.stringify({ refreshToken }),
+    });
+
+    if (response.ok) {
+      // Clear access token and refresh token from local storage (or wherever you've stored them)
+      localStorage.removeItem('accessToken');
+      localStorage.removeItem('refreshToken');
+      return true;
+    } else {
+      console.error('Logout failed', await response.text());
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/api/requests/logout.jsx:18]: Remove log

File: prediction-polls/frontend/src/api/requests/logout.jsx [hunk 6]
```diff
@@ -0,0 +1,27 @@
...
+      return false;
+    }
+  } catch (error) {
+    console.error('Logout error', error);
```
Comment (YigitSekerci) [inline prediction-polls/frontend/src/api/requests/logout.jsx:22]: Remove log

**Review by YigitSekerci — APPROVED** [review]
I guess you should remove console logs but other than that it looks okey.

**Review by EmreBatuhan — APPROVED** [review]
This code successfully ran on our current deployment as intended. I also agree @YigitSekerci on deleting unnecessary logs but other than that it is good.

**Pull request merged**